lib/spack/spack/util/git.py

```diff
@@ -27,15 +29,36 @@ def _find_git() -> Optional[str]:
     return exe.which_string("git", required=False)
 
 
+def extract_git_version_str(git_exe):
+    v_str = git_exe("--version", output=str).strip().split()[-1].replace("windows.", "")
+    return v_str
+
+
+class GitExecutable(exe.Executable):
+    def __init__(self, name=None):
+        if not name:
+            name = _find_git()
```

Comment:
Is the `required=False` behavior of `_find_get` correct here? I would've thought it needs to be `required=True` at this point.